core/src/main/scala/com/salesforce/op/dsl/RichDateFeature.scala

```diff
@@ -183,3 +185,40 @@ trait RichDateFeature {
   }
 
 }
+
+abstract class TimePeriodTransformers[T <: Date : TypeTag] {
+  val f: FeatureLike[T]
+
+  /**
+   * Converts a Date or DateTime feature into a selected time period.
+   *
+   * @param period type of [[TimePeriod]] to convert date feature to
+   * @return Integer feature of time period value
+   */
+  def toTimePeriod(period: TimePeriod): FeatureLike[Integral] = {
+    val periodFun: Long => Int = period match {
+      case TimePeriod.DayOfMonth => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).dayOfMonth.get
+      case TimePeriod.DayOfWeek => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).dayOfWeek.get
+      case TimePeriod.DayOfYear => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).dayOfYear.get
+      case TimePeriod.HourOfDay => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).hourOfDay.get
+      case TimePeriod.MonthOfYear => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).monthOfYear.get
+      case TimePeriod.WeekOfMonth => t => {
+        val dt = new JDateTime(t, DateTimeUtils.DefaultTimeZone)
+        dt.weekOfWeekyear.get - dt.withDayOfMonth(1).weekOfWeekyear.get
+      }
+      case TimePeriod.WeekOfYear => t => new JDateTime(t, DateTimeUtils.DefaultTimeZone).weekOfWeekyear.get
+    }
+    f.transformWith(
+      new UnaryLambdaTransformer[T, Integral](operationName = "dateToTimePeriod",
+        transformFn = _.value.map(t => periodFun(t).toLong).toIntegral)
+    )
+  }
+
+  def toDayOfMonth(): FeatureLike[Integral] = toTimePeriod(TimePeriod.DayOfMonth)
```

Comment:
Is there much value of having these methods? let's just have `def toTimePeriod(period: TimePeriod)` and that way you wont need this abstract class bu simply add `toTimePeriod` to both `RichDateTimeFeature` and `RichDateFeature`.

Reply:
I think so. As a user, I would much prefer to use these shortcuts and likely not use the time period shortcuts at all.

Reply:
Once introduced, we would have to maintain them ;) and each time a new entry in enum is added we would need to add a new method. that's why I just prefer a single method with an enum argument.